File: 2645-pass-the-pillow/2645-pass-the-pillow.go
func passThePillow(n int, time int) int {
    pos , dir := 0 , 1
    for i:= 0 ; i<time ;i++{
        if dir == 1 {
            if pos == n-1 {
                dir =-1 
            }
            pos += dir
        }else{
            if pos == 0 {
                dir = 1
            }
            pos += dir
        }
    }
    return pos +1
}

File: 2645-pass-the-pillow/2645-pass-the-pillow.java
class Solution {
    public int passThePillow(int n, int time) {
  int cycleLength = (n - 1) * 2;
        int cyclePosition = time % cycleLength;
        if (cyclePosition < n) {
            return cyclePosition + 1;
        } else {
            return cycleLength - cyclePosition + 1;
        }
    }
}